File: cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float dollars;
    // Prompt user to enter change owed.
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0);

    // Convert float to int for calculation
    int cents = round(dollars * 100);

    int count = 0;

    // Increase the count by one for every change dispensed.
    while (cents >= 25)
    {
        cents -= 25;
        count++;
    }
    while (cents >= 10)
    {
        cents -= 10;
        count++;
    }
    while (cents >= 5)
    {
        cents -= 5;
        count++;
    }
    while (cents >= 1)
    {
        cents -= 1;
        count++;
    }

    // Print the number of least coins needed.
    printf("%i\n", count);
}

File: pset6- hello, mario less, mario more, cash, credit, readability, dna/cash/cash.py
from cs50 import get_float

# Get user input
change = get_float("Change owed: ")

# Validate user's input
while (change < 0):
    change = get_float("Change owed: ")

count = 0

# Provide 25 cent if change is than 0.25
while (change >= 0.25):
    count += 1
    change = round(change - 0.25, 2)

# Provide 10 cent if change is than 0.10
while (change >= 0.10):
    count += 1
    change = round(change - 0.10, 2)

# Provide 5 cent if change is than 0.05
while (change >= 0.05):
    count += 1
    change = round(change - 0.05, 2)

# Provide 1 cent if change is than 0.01
while (change >= 0.01):
    count += 1
    change = round(change - 0.01, 2)

print(count)
